### DISCUSSION TITLE
Not sure how to leverage custom devcontainer definitions

### DISCUSSION THREAD

**Discussion by sheldonhull** [community/community#28222] [Codespaces] [answered]
I've read the docs and find the concept of custom or extending current devcontainers confusing.

Basically I want to leverage the default `mcr.microsoft.com/vscode/devcontainers/universal` but ensure a few tweaks to env variables, and a few additional tools are installed.

The docs aren't clear on:

- How can I provide a custom entry in the devcontainers listed in the drop down to add configuration files?
- Or how can I extend a current definition so that the additional features and scripts I want to run will be installed. 
   - Part of that is not sure how I can add my own script-library features so setup scripts I create can easily be reused in new workspaces.

**Comment by jshorty** [comment, marked as answer]
Hi there! Have you already consulted the steps here?
https://github.com/microsoft/vscode-dev-containers/tree/main/containers/codespaces-linux#adding-the-definition-to-a-project-or-codespace

If you just want to use this devcontainer but with minor tweaks, you could also do further custom setup within a [lifecycle script](https://code.visualstudio.com/docs/remote/devcontainerjson-reference#_lifecycle-scripts).

**Reply by sheldonhull** [reply]
I don't think this answers the usage of overriding the default container definition with a local based on path, but I've since moved away from even trying that, instead I'm using the features preview and it works great. I'm marking as answer since this solves the issue I needed in the first place.
